lib/improver/generate_ancillaries/generate_ancillary.py

```diff
@@ -0,0 +1,251 @@
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing ancillary generation utilities for Improver"""
+
+import iris
+import numpy as np
+from glob import glob
+
+
+def _make_mask_cube(mask_data, key, coords,
+                    upper_threshold=None, lower_threshold=None):
+    """
+    Makes cube from numpy masked array generated from orography fields.
+
+    Parameters
+    ----------
+    mask_data : numpy masked array
+        The numpy array to make a cube from.
+    key : string
+        Key from THRESHOLD_DICT which descibes type of topography band.
+    coords : dictionary
+        Dictionary of coordinate on the model ancillary file.
+    upper_threshold : float
+        Upper topographic bound defining the mask.
+    lower_threshold : float
+        Lower topographic bound defining the mask.
+
+    Returns
+    -------
+    mask_cube : cube
+        Cube containing the mask_data array, with appropriate coordinate
+        and attribute information.
+    """
+    mask_cube = iris.cube.Cube(mask_data, long_name='Topography mask')
+    if lower_threshold or lower_threshold == 0:
+        coord_name = 'topographic_bound_lower'
+        threshold_coord = iris.coords.AuxCoord(lower_threshold,
+                                               long_name=coord_name)
+        mask_cube.add_aux_coord(threshold_coord)
+    if upper_threshold or upper_threshold == 0:
+        coord_name = 'topographic_bound_upper'
+        threshold_coord = iris.coords.AuxCoord(upper_threshold,
+                                               long_name=coord_name)
+        mask_cube.add_aux_coord(threshold_coord)
+    mask_cube.attributes['Topographical Type'] = key.title()
+    for coord in coords:
+        if coord.name() in ['projection_y_coordinate', 'latitude']:
+            mask_cube.add_dim_coord(coord, 0)
+        elif coord.name() in ['projection_x_coordinate', 'longitude']:
+            mask_cube.add_dim_coord(coord, 1)
+        else:
+            mask_cube.add_aux_coord(coord)
+    return mask_cube
+
+
+def find_standard_ancil(grid, stage_ancil):
+    """
+    Reads standard ancillary from StaGE or raises exception.
+
+    Parameters
+    -----------
+    grid : string
+      String describing standard grid
+
+    stage_ancil : string
+      Location of stage ancillaries
+
+    Returns
+    --------
+    standard_ancil : cube
+        The StaGE ancil cube.
+
+    Raises
+    -------
+    IOError: if input ancillary cannot be found in either StaGE or UM
+             directories.
```

Comment:
This comment is out of date, I previously checked for UM ancils which would be regridded, but that functionality has been removed and the code now assumes StaGE will provide the input. I've updated the comment accordingly.